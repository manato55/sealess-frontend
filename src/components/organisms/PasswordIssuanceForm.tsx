import React, { Dispatch, SetStateAction, useCallback, useState, useEffect } from 'react';
import Button from '../atoms/Button';
import FormWrapper from '../atoms/FormWrapper';
import ErrorMessageWrapper from '../atoms/ErrorMessageWrapper';
import Input from '../atoms/Input';
import { useAuthenticate } from '../../hooks/useAuth';
import { useRecoilValue, useRecoilState } from 'recoil';
import { authErrorMessage, eachErrorFlag } from '../../store/atom';
import styled from 'styled-components';
import Link from 'next/link';

interface Props {
  email: string;
  setEmail: Dispatch<SetStateAction<string>>;
}

export const PasswordIssuanceForm = (props: Props): React.ReactElement => {
  const errorMessage = useRecoilValue(authErrorMessage);
  const [errorFlag, setErrorFlag] = useRecoilState(eachErrorFlag);
  const { passwordReRegister } = useAuthenticate();
  const [message, setMessage] = useState<string>('');

  const emailHandler = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>): void => {
      props.setEmail(e.target.value);
    },
    [props]
  );

  const handleSubmit = useCallback(
    async (e: React.ChangeEvent<HTMLFormElement>) => {
      setMessage('');
      e.preventDefault();
      if (!confirm('送信しますか？')) {
        return;
      }
      const res = await passwordReRegister(props.email);
      if (res === 200) {
        setMessage('リンクを付けたメールを送信しました。');
      }
    },
    [props, passwordReRegister]
  );

  return (
    <>
      <FormWrapper>
        {message}
        <ErrorMessageWrapper>{errorFlag.email && errorMessage}</ErrorMessageWrapper>
        <form onSubmit={handleSubmit}>
          <Input
            type="email"
            value={props.email}
            placeholder={'メールアドレス'}
            onChange={(e) => emailHandler(e)}
          />
          <Button type="submit">送信</Button>
        </form>
        <LoginLink>
          <Link href="/login" passHref>
            <A>ログイン画面</A>
          </Link>
        </LoginLink>
      </FormWrapper>
    </>
  );
};

const LoginLink = styled.div`
  margin-top: 10px;
  text-align: left;
`;

const A = styled.a`
  text-decoration: underline;
  color: blue;
`;

export default React.memo(PasswordIssuanceForm);
