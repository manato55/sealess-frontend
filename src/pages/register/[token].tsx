import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
