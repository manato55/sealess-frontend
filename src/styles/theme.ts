const theme = {
  palette: {
    primary: {
      main: '#09869a',
      contrastText: '#fff',
    },
    secondary: {
      main: '#03182b',
      contrastText: '#fff',
    },
    lightGray: {
      main: '#efefef',
      dark: '#cfcfcf',
      contrastText: '#333',
    },
    black: {
      main: '#333',
      contrastText: '#fff',
    },
    white: {
      main: '#fff',
      contrastText: '#333',
    },
  },
  text: {
    primary: '#333',
    white: '#fff',
    placeholder: '#757575',
    disabled: '#c0c0c0',
  },
  layout: {
    pc: {
      containerWidth: '1280px',
      headerHeight: '80px',
      headerDenseHeight: '56px',
      footerHeight: '250px',
      heroHeight: '420px',
      pankuzuHeight: '40px',
      content: {
        titleHeight: '240px',
        width: '1280px',
        xPadding: '20px',
      },
      headerFixedPos: 200,
    },
    sp: {
      containerWidth: '375px',
      headerHeight: '60px',
      headerDenseHeight: '60px',
      footerHeight: '40px',
      heroHeight: '230px',
      pankuzuHeight: '20px',
      content: {
        titleHeight: '120px',
        width: '100%',
        xPadding: '20px',
      },
      headerFixedPos: 100,
    },
  },
  breakpoints: {
    sp: 640,
    pcSm: 1280,
    pc: Infinity,
  },
};

export default theme;
