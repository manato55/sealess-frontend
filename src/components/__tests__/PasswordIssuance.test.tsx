import React from 'react';
import PasswordIssuanceForm from '../organisms/PasswordIssuanceForm';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles/theme';

describe('PasswordIssuanceForm test', () => {
  let container;
  const onChangeMock = jest.fn();

  beforeEach(() => {
    container = mount(
      <RecoilRoot>
        <ThemeProvider theme={theme}>
          <PasswordIssuanceForm setEmail={onChangeMock} email={'test@test.com'} />
        </ThemeProvider>
      </RecoilRoot>
    );
  });

  test('input test', () => {
    const input = container.find('input');
    input.simulate('change');
    expect(onChangeMock).toHaveBeenCalledWith('test@test.com');
  });

  test('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
