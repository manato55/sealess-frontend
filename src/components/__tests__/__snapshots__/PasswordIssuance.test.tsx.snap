// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordIssuanceForm test snapshot 1`] = `
<RecoilRoot>
  <RecoilRoot_INTERNAL>
    <Batcher
      setNotifyBatcherOfChange={[Function]}
    />
    <Component
      theme={
        Object {
          "breakpoints": Object {
            "pc": Infinity,
            "pcSm": 1280,
            "sp": 640,
          },
          "layout": Object {
            "pc": Object {
              "containerWidth": "1280px",
              "content": Object {
                "titleHeight": "240px",
                "width": "1280px",
                "xPadding": "20px",
              },
              "footerHeight": "250px",
              "headerDenseHeight": "56px",
              "headerFixedPos": 200,
              "headerHeight": "80px",
              "heroHeight": "420px",
              "pankuzuHeight": "40px",
            },
            "sp": Object {
              "containerWidth": "375px",
              "content": Object {
                "titleHeight": "120px",
                "width": "100%",
                "xPadding": "20px",
              },
              "footerHeight": "40px",
              "headerDenseHeight": "60px",
              "headerFixedPos": 100,
              "headerHeight": "60px",
              "heroHeight": "230px",
              "pankuzuHeight": "20px",
            },
          },
          "palette": Object {
            "black": Object {
              "contrastText": "#fff",
              "main": "#333",
            },
            "lightGray": Object {
              "contrastText": "#333",
              "dark": "#cfcfcf",
              "main": "#efefef",
            },
            "primary": Object {
              "contrastText": "#fff",
              "main": "#09869a",
            },
            "secondary": Object {
              "contrastText": "#fff",
              "main": "#03182b",
            },
            "white": Object {
              "contrastText": "#333",
              "main": "#fff",
            },
          },
          "text": Object {
            "disabled": "#c0c0c0",
            "placeholder": "#757575",
            "primary": "#333",
            "white": "#fff",
          },
        }
      }
    >
      <Memo(PasswordIssuanceForm)
        email="test@test.com"
        setEmail={
          [MockFunction] {
            "calls": Array [
              Array [
                "test@test.com",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      >
        <styled.div>
          <div
            className="sc-gtsrHT ixkiJv"
          >
            <styled.p>
              <p
                className="sc-dlnjwi iUWSvF"
              />
            </styled.p>
            <form
              onSubmit={[Function]}
            >
              <styled.input
                editable={true}
                onChange={[Function]}
                placeholder="メールアドレス"
                type="email"
                value="test@test.com"
                visible={false}
              >
                <input
                  className="sc-hKFxyN fhRRGw"
                  onChange={[Function]}
                  placeholder="メールアドレス"
                  type="email"
                  value="test@test.com"
                />
              </styled.input>
              <styled.button
                background="dark"
                marginTop={40}
                type="submit"
              >
                <button
                  className="sc-bdnxRM gkxubh"
                  type="submit"
                >
                  送信
                </button>
              </styled.button>
            </form>
          </div>
        </styled.div>
      </Memo(PasswordIssuanceForm)>
    </Component>
  </RecoilRoot_INTERNAL>
</RecoilRoot>
`;
